### PR TITLE
Get DOMAIN from api pod for upgrade

### PR DESCRIPTION
**Pull request by HartS**

### CODE REVIEW

**Comment by HartS** [comment]
Passed in pipeline

File: qa-pipelines/tasks/cf-upgrade.sh [hunk 1]
```diff
@@ -6,20 +6,19 @@ mkdir -p /root/.kube/
...
-# Domain for SCF. DNS for *.DOMAIN must point to the kube node's
-# external ip. This must match the value passed to the
-# cert-generator.sh script.
-DOMAIN=${external_ip}.${MAGIC_DNS_SERVICE}
+CF_NAMESPACE=scf
+UAA_NAMESPACE=uaa
+# Domain for SCF, taken from the api pod
+DOMAIN=$(kubectl get pods -o json --namespace ${CF_NAMESPACE} api-0 | jq -r  '.spec.containers[0].env[] |  select(.name == "DOMAIN").value')
+api_prefix=${DOMAIN%.${MAGIC_DNS_SERVICE}}
+external_ip=${api_prefix#api.}
```
Comment (prabalsharma) [inline qa-pipelines/tasks/cf-upgrade.sh]: why do you need api prefix for external ip?

File: qa-pipelines/tasks/cf-upgrade.sh [hunk 2]
```diff
@@ -6,20 +6,19 @@ mkdir -p /root/.kube/
...
-external_ip=$(kubectl get nodes -o json | jq -r '.items[] | select(.spec.unschedulable == true | not) | .metadata.annotations["alpha.kubernetes.io/provided-node-ip"]' | head -n1)
-# Domain for SCF. DNS for *.DOMAIN must point to the kube node's
-# external ip. This must match the value passed to the
-# cert-generator.sh script.
-DOMAIN=${external_ip}.${MAGIC_DNS_SERVICE}
+CF_NAMESPACE=scf
+UAA_NAMESPACE=uaa
+# Domain for SCF, taken from the api pod
+DOMAIN=$(kubectl get pods -o json --namespace ${CF_NAMESPACE} api-0 | jq -r  '.spec.containers[0].env[] |  select(.name == "DOMAIN").value')
+api_prefix=${DOMAIN%.${MAGIC_DNS_SERVICE}}
```
Comment (prabalsharma) [inline qa-pipelines/tasks/cf-upgrade.sh]: could have been external_ip

File: qa-pipelines/tasks/cf-upgrade.sh (hunk 1 above)
Comment (prabalsharma) [inline qa-pipelines/tasks/cf-upgrade.sh]: no api prefix to remove

File: qa-pipelines/tasks/cf-upgrade.sh [hunk 3]
```diff
@@ -6,20 +6,19 @@ mkdir -p /root/.kube/
...
+external_ip=${api_prefix#api.}
+unset api_prefix
```
Comment (prabalsharma) [inline qa-pipelines/tasks/cf-upgrade.sh]: we already have `set +o allexport` below

**Pull request merged**